gotham/src/router/mod.rs

```diff
@@ -108,7 +109,13 @@ impl Handler for Router {
                         },
                         Err(status) => {
                             trace!("[{}] responding with error status", request_id(&state));
-                            let res = create_response(&state, status, None);
+                            let mut res = create_response(&state, status, None);
+
+                            if let StatusCode::MethodNotAllowed = status {
```

Comment:
I'm half wondering if this should be generic in some capacity `Node` would have a mapping on headers to set for each status code, might be overkill...

Reply:
The thought definitely crossed my mind, but seemed like over-engineering and something we could add later if there's another case.

If you feel strongly about it, I can address it now. No big deal.

Reply:
Lets wait for a use case, consider this resolved for now.